odl/deform/linearized.py

```diff
@@ -0,0 +1,545 @@
+# Copyright 2014-2016 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Operators and functions for linearized deformation."""
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+standard_library.install_aliases()
+
+from odl.operator.operator import Operator
+from odl import ProductSpace, DiscreteLp, Gradient, PointwiseInner, Divergence
+import numpy as np
+
+
+__all__ = ('LinDeformFixedTempl', 'LinDeformFixedTemplDeriv',
+           'LinDeformFixedTemplDerivAdj', 'LinDeformFixedDisp',
+           'LinDeformFixedDispAdj', 'linear_deform', 'deform_grad')
+
+
+def linear_deform(template, displacement):
```

Comment:
How about we define two: _linear_deform (for private) and linear_deform(for public)? If so, maybe more efficient.